### PR TITLE
Scrapper, Sofia Onofrienko - 22FPL1

### PR DESCRIPTION
**Pull request by soniaonofrienko**

### CODE REVIEW

File: lab_5_scrapper/scrapper.py [hunk 1]
```diff
@@ -18,6 +27,11 @@ def __init__(self, path_to_config: pathlib.Path) -> None:
         Args:
             path_to_config (pathlib.Path): Path to configuration.
         """
+        self.CRAWLER_CONFIG_PATH = path_to_config
+        self.ConfigDTO = self._extract_config_content()
```
Comment (marina-kaz) [inline lab_5_scrapper/scrapper.py:31]: 1. Please do not name attribute with camel case
2. Why public?

File: lab_5_scrapper/scrapper.py [hunk 2]
```diff
@@ -26,11 +40,28 @@ def _extract_config_content(self) -> ConfigDTO:
...
+        config_DTO = ConfigDTO(**config)
+        return config_DTO
 
     def _validate_config_content(self) -> None:
         """
         Ensure configuration parameters are not corrupt.
         """
+        if not all(seed.startswith('https://antropogenez.ru/news/') for seed in self.config_DTO.seed_urls):
+            raise Exception('IncorrectSeedURLError')
+        if self.config_DTO.total_articles < 1 or self.config_DTO.total_articles > 150:
```
Comment (marina-kaz) [inline lab_5_scrapper/scrapper.py:53]: Please use pre-defined constants from `core_utils` to specify limits

File: lab_5_scrapper/scrapper.py [hunk 3]
```diff
@@ -26,11 +40,28 @@ def _extract_config_content(self) -> ConfigDTO:
...
+            raise Exception('NumberOfArticlesOutOfRangeError')
+        if not isinstance(self.config_DTO.total_articles, int):
+            raise Exception('IncorrectNumberOfArticlesError')
+        if not isinstance(self.config_DTO.headers, dict):
+            raise Exception('IncorrectHeadersError')
+        if not isinstance(self.config_DTO.encoding, str):
+            raise Exception('IncorrectEncodingError')
+        if self.config_DTO.timeout < 1 or self.config_DTO.timeout > 60:
```
Comment (marina-kaz) [inline lab_5_scrapper/scrapper.py:61]: Here as well

**Review by marina-kaz — COMMENTED** [review]
Well, it's a start, keep going!
Please do not comment code that wasn't written by you.

File: lab_5_scrapper/scrapper.py (hunk 2 above)
Comment (marina-kaz) [reply lab_5_scrapper/scrapper.py]: image

**Comment by marina-kaz** [comment]
Little changes since my last review, no comments were addressed

File: lab_5_scrapper/scrapper.py (hunk 2 above)
Comment (soniaonofrienko) [reply lab_5_scrapper/scrapper.py]: done

File: lab_5_scrapper/scrapper.py (hunk 3 above)
Comment (soniaonofrienko) [reply lab_5_scrapper/scrapper.py]: done

File: lab_5_scrapper/scrapper.py [hunk 4]
```diff
@@ -19,18 +75,55 @@ def __init__(self, path_to_config: pathlib.Path) -> None:
             path_to_config (pathlib.Path): Path to configuration.
         """
 
+        self.path_to_config = path_to_config
+        self._validate_config_content()
+        self.config = self._extract_config_content()
```
Comment (marina-kaz) [inline lab_5_scrapper/scrapper.py:80]: Why store both instance and all its properties as attributes? Isn't it redundant?

File: lab_5_scrapper/scrapper.py [hunk 5]
```diff
@@ -100,6 +200,14 @@ def make_request(url: str, config: Config) -> requests.models.Response:
...
         requests.models.Response: A response from a request
     """
+    sleep(random.randrange(3))
+
+    return requests.get(
+        url=url,
+        timeout=config.get_timeout(),
+        headers=config.get_headers(),
+        verify=config.get_verify_certificate()
+    )
```
Comment (marina-kaz) [inline lab_5_scrapper/scrapper.py:210]: Make sure to also use `encoding` provided in `scrapper_config.json`

File: lab_5_scrapper/scrapper.py [hunk 6]
```diff
@@ -116,6 +224,9 @@ def __init__(self, config: Config) -> None:
         Args:
             config (Config): Configuration
         """
+        self.config = config
```
Comment (marina-kaz) [inline lab_5_scrapper/scrapper.py:227]: Why public?

**Review by marina-kaz — COMMENTED** [review]
Good job!
Maybe try for mark 10 to make up for late submission?

**Pull request closed without merging**